### PR TITLE
perf: Remove network calls to validate the URLs on main thread

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Removing network calls on main thread.
This is a potential fix for https://github.com/status-im/status-desktop/issues/15273

The URL validator needs to be refactored to use an async connection.

### CODE REVIEW

**Comment by alexjba** [comment]
@anastasiyaig @Valentina1133 This is a potential fix. Could you please test it out? It's based on 2.29.x

**Comment by Valentina1133** [comment]
> @anastasiyaig @Valentina1133 This is a potential fix. Could you please test it out? It's based on 2.29.x

I tested it, seems like all good now, the app doesn't become unresponsive after fast clicking between tabs

**Comment by anastasiyaig** [comment]
it works now, i will launch my autotests to confirm  as well but so far it looks very good

https://github.com/status-im/status-desktop/assets/82375995/e1b5bbca-bb91-4c7d-a340-93f5e7b864c8

**Comment by alexjba** [comment]
@Khushboo-dev-cpp I'm thinking on removing the URL validation for now since it's validating URLs on main thread. Is it ok with you if I create another task to validate the URLs async?

On the other hand IMO we don't need to validate these urls..Or at least not in this form. Doesn't really make sense to me to warn the user before opening a url by click and let him choose not to expose himself to the URL, but on the other had to have a feature where the URL is automatically called.

If we need to validate the urls I'd opt for a bulletproof approach:
1. Use async validation
2. Validate only if the url/domain has been whitelisted by the user

WDYT?

File: ui/imports/utils/Utils.qml [hunk 1]
```diff
@@ -961,6 +961,6 @@ QtObject {
     }
 
     function getUrlStatus(url) {
-        return globalUtilsInst.isValidURL(url)
+        return true
```
Comment (caybro) [inline ui/imports/utils/Utils.qml:964]: `bool QUrl::isValid() const` just saying :)

**Review by caybro — APPROVED** [review]

File: ui/imports/utils/Utils.qml (hunk 1 above)
Comment (alexjba) [reply ui/imports/utils/Utils.qml:964]: this validator checks if the HTTP `get` call returns 200. `QUrl::isValid` can't really replace it.

Comment (caybro) [reply ui/imports/utils/Utils.qml:964]: ah, so it does a connectivity check... hmm

Comment (caybro) [reply ui/imports/utils/Utils.qml:964]: We have that facility in DOtherSide too

**Review by anastasiyaig — APPROVED** [review]

**Comment by Khushboo-dev-cpp** [comment]
> @Khushboo-dev-cpp I'm thinking on removing the URL validation for now since it's validating URLs on main thread. Is it ok with you if I create another task to validate the URLs async?
> 
> On the other hand IMO we don't need to validate these urls..Or at least not in this form. Doesn't really make sense to me to warn the user before opening a url by click and let him choose not to expose himself to the URL, but on the other had to have a feature where the URL is automatically called.
> 
> If we need to validate the urls I'd opt for a bulletproof approach:
> 
> 1. Use async validation
> 2. Validate only if the url/domain has been whitelisted by the user
> 
> WDYT?

@alexjba this was added for use in the CollectibleDetailView where we let the user go to opensea or the explorer to go check their collectible. @benjthayer  wanted to avoid users reach a 404 not found error when the link is clicked.

If this fix it solving the issue, we can go ahead and remove it, but please add a TODO with a task number in the code first! else all of this conversation is lost. 

And yes calling an async api would be a great idea! We can talk to design and see if they really dont want this to happen. Can be discussed in the other issue you create!

File: ui/imports/utils/Utils.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/utils/Utils.qml:964]: add TODO with link to GH issue created to make a decision if its needed or to make the call async

Comment (caybro) [reply ui/imports/utils/Utils.qml:964]: I wonder how making the backend call async will help with calling it from a plain JS function

Comment (caybro) [reply ui/imports/utils/Utils.qml:964]: I mean, you will never get any result this way

Comment (alexjba) [reply ui/imports/utils/Utils.qml:966]: A refactoring is needed. We'll need the async support in qml as well

**Review by caybro — APPROVED** [review]

**Pull request merged**